Client/Frontend/Browser/Onboarding/OnboardingNavigationController.swift

```diff
@@ -0,0 +1,121 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import Shared
+
+private let log = Logger.browserLogger
+
+protocol Onboardable: class {
+    /// Show next on boarding screen if possible.
+    /// If last screen is currently presenting, the view is dimissed instead(onboarding finished).
+    func presentNextScreen(current: OnboardingViewController)
+    /// Skip all onboarding screens, onboarding is considered as completed.
+    func skip()
+}
+
+protocol OnboardingControllerDelegate: class {
+    func onboardingCompleted(_ onboardingController: OnboardingNavigationController)
+}
+
+class OnboardingNavigationController: UINavigationController {
+    
+    private struct UX {
+        /// The onboarding screens are showing as a modal on iPads.
+        static let preferredModalSize = CGSize(width: 375, height: 667)
+    }
+    
+    weak var onboardingDelegate: OnboardingControllerDelegate?
+    
+    enum Screens: CaseIterable {
+        case searchEnginePicker
+        case shieldsInfo
+        
+        func viewController(with profile: Profile) -> OnboardingViewController {
+            switch self {
+            case .searchEnginePicker:
+                return OnboardingSearchEnginesViewController(profile: profile)
+            case .shieldsInfo:
+                return OnboardingShieldsViewController(profile: profile)
+            }
+        }
+        
+        var type: AnyClass {
+            switch self {
+            case .searchEnginePicker: return OnboardingSearchEnginesViewController.self
+            case .shieldsInfo: return OnboardingShieldsViewController.self
+            }
+        }
+    }
+    
+    init?(profile: Profile) {
+        guard let firstScreen = Screens.allCases.first else { return nil }
+        
+        let firstViewController = firstScreen.viewController(with: profile)
+        super.init(rootViewController: firstViewController)
+        firstViewController.delegate = self
+        
+        isNavigationBarHidden = true
+        if UIDevice.current.userInterfaceIdiom == .phone {
```

Comment:
Just a qq, do you prefer this over ternary?

` modalPresentationStyle = UIDevice.current.userInterfaceIdiom == .phone ? .fullScreen : .formSheet`

Reply:
previously this condition included iOS 13 check and I forgot to simplify it after I removed it, thanks